### ISSUE TITLE
Use ghc-lib to support a range of GHC versions

### ISSUE THREAD

**Issue by arybczak** [mpickering/apply-refact#57]
Did you try to make it work?

Simply changing ghc dependency to ghc-lib doesn't work because of ghc-exactprint type mismatches, so presumably ghc-exactprint would also need to use ghc-lib.

**Comment by zliu41** [comment]
Yeah, I don't see how apply-refact can migrate to ghc-lib unless ghc-exactprint migrates first.

**Comment by ndmitchell** [comment]
ghc-exactprint would definitely need to migrate (or offer the option) first.

If ghc-exactprint and apply-refact were on matching versions of ghc-lib, I'd be slightly tempted to wonder if we could link in apply-refact as a library into HLint.

**Comment by zliu41** [comment]
Sounds good; I raised https://github.com/alanz/ghc-exactprint/issues/93.
Not sure how much work it is; I may have some time to work on it, but my time available for open source work over the next few months will be very limited.

**Comment by arybczak** [comment]
Will that be possible with 9.2 now that `ghc-exactprint` was integrated into GHC?
